components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/utils/BPMNDataReceiverConfig.java

```diff
@@ -0,0 +1,99 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.bpmn.analytics.publisher.utils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.bpmn.analytics.publisher.AnalyticsPublisherConstants;
+import org.wso2.carbon.bpmn.analytics.publisher.internal.BPMNAnalyticsHolder;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.api.Resource;
+import org.wso2.carbon.user.api.UserStoreException;
+
+/**
+ * BPMNDataReceiverConfig is used by AnalyticsPublisher to retrieve user name and password
+ */
+public class BPMNDataReceiverConfig {
+	private static Log log = LogFactory.getLog(BPMNDataReceiverConfig.class);
+
+	/**
+	 * Get thrift url of data receiver
+	 *
+	 * @return thrift url of data receiver
+	 * @throws RegistryException
+	 */
+	public static String getThriftURL() throws RegistryException {
+		String url;
+		Registry registry =
+				BPMNAnalyticsHolder.getInstance().getRegistryService().getConfigSystemRegistry();
+
+		if (registry.resourceExists(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH)) {
+			Resource resource =
+					registry.get(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH);
+			url = resource.getProperty(AnalyticsPublisherConstants.THRIFT_URL_PROPERTY);
+		}else{
+			url = AnalyticsPublisherConstants.LOCAL_THRIFT_URL;
+		}
+
+		return url;
+	}
+
+	/**
+	 * Get user name
+	 *
+	 * @return user name
+	 * @throws RegistryException
+	 * @throws UserStoreException
+	 */
+	public static String getUserName() throws RegistryException, UserStoreException {
+		String userName;
+		Registry registry =
+				BPMNAnalyticsHolder.getInstance().getRegistryService().getConfigSystemRegistry();
+
+		if (registry.resourceExists(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH)) {
+			Resource resource =
+					registry.get(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH);
+			userName = resource.getProperty(AnalyticsPublisherConstants.USER_NAME_PROPERTY);
+		}else{
+			userName = AnalyticsPublisherConstants.USER_NAME;
+		}
+		return userName;
+	}
+
+	/**
+	 * Get password
+	 *
+	 * @return password
+	 * @throws RegistryException
+	 * @throws UserStoreException
+	 */
+	public static String getPassword() throws RegistryException, UserStoreException {
+		String password;
+		Registry registry =
+				BPMNAnalyticsHolder.getInstance().getRegistryService().getConfigSystemRegistry();
+
+		if (registry.resourceExists(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH)) {
+			Resource resource =
+					registry.get(AnalyticsPublisherConstants.DATA_RECEIVER_RESOURCE_PATH);
+			password = resource.getProperty(AnalyticsPublisherConstants.PASSWORD_PROPERTY);
+		}else{
```

Comment:
These methods were updated according to the latest data pblisher. Already fixed.